config/config_helpers.go

```diff
@@ -32,22 +33,52 @@ func ResolveTerragruntConfigString(terragruntConfigString string, include *Inclu
 // Resolve a single call to an interpolation function of the format ${some_function()} in a .terragrunt file
 func resolveTerragruntInterpolation(str string, include *IncludeConfig, terragruntOptions *options.TerragruntOptions) (string, error) {
 	matches := HELPER_FUNCTION_SYNTAX_REGEX.FindStringSubmatch(str)
-	if len(matches) == 2 {
-		return executeTerragruntHelperFunction(matches[1], include, terragruntOptions)
+	if len(matches) == 3 {
+		return executeTerragruntHelperFunction(matches[1], matches[2], include, terragruntOptions)
 	} else {
 		return "", errors.WithStackTrace(InvalidInterpolationSyntax(str))
 	}
 }
 
 // Execute a single Terragrunt helper function and return its value as a string
-func executeTerragruntHelperFunction(functionName string, include *IncludeConfig, terragruntOptions *options.TerragruntOptions) (string, error) {
+func executeTerragruntHelperFunction(functionName string, parameters string, include *IncludeConfig, terragruntOptions *options.TerragruntOptions) (string, error) {
 	switch functionName {
 	case "find_in_parent_folders": return findInParentFolders(terragruntOptions)
 	case "path_relative_to_include": return pathRelativeToInclude(include, terragruntOptions)
+	case "get_env": return getEnvironmentVariables(parameters, terragruntOptions)
 	default: return "", errors.WithStackTrace(UnknownHelperFunction(functionName))
 	}
 }
 
+func extractParametersFromFunction(parameters string) (map[string]string, error) {
+	matches := HELPER_FUNCTION_GET_ENV_PARAMETERS_SYNTAX_REGEX.FindStringSubmatch(parameters)
+	if (len(matches)) < 2 {
+		return nil, errors.WithStackTrace(InvalidFunctionParameters(parameters))
+	}
+	result := make(map[string]string)
+	for value, name := range HELPER_FUNCTION_GET_ENV_PARAMETERS_SYNTAX_REGEX.SubexpNames() {
+		if value != 0 {
+			result[name] = matches[value]
+		}
+  }
+
+	return result, nil
+}
+
+func getEnvironmentVariables(parameters string, terragruntOptions *options.TerragruntOptions) (string, error) {
+	parameterMap, err := extractParametersFromFunction(parameters)
```

Comment:
`extractParametersFromFunction` makes it sound like a generic function that could parse any type of parameter, but the reality is that the function uses a specific regex to parse a specific type of parameter (an environment variable and a default fallback value). To use the opaque `map[string]string` that it returns, you have to know what regex was used. I found this a little confusing.

Perhaps a cleaner approach is to:

1. Rename the function to `parseGetEnvParameters`.
1. Instead of returning a `map[string]string`, have the function return a struct with clearly named fields:
    ```go
    type EnvVar struct {
      Name         string
      DefaultValue string
    }
    ```

This way, all the regex details are fully encapsulated within the function, and it's easier to see what the function does and how to use its return value.